cmd/init.go

```
@@ -143,7 +143,10 @@ to quickly create a Cobra application.`,
 		// todo: this doesn't default to testing the dns check
 		skipHostedZoneCheck := viper.GetBool("init.hostedzonecheck.enabled")
 		if !skipHostedZoneCheck {
-			aws.TestHostedZoneLiveness(globalFlags.DryRun, awsFlags.HostedZoneName, hostedZoneId)
+			hostedZoneLiveness := aws.TestHostedZoneLiveness(dryRun, hostedZoneName, hostedZoneId)
+			if !hostedZoneLiveness {
+				log.Panic("Fail to check the Liveness of HostedZone, we need a valid public HostedZone on the same AWS account that Kubefirst will be installed.")
+			}
```

This should return a Err, now that init support it.